### ISSUE TITLE
Add the possibility of marking time stamps

### ISSUE THREAD

**Issue by Dapid** [tensorflow/tensorboard#411]
It would be nice if it were possible to mark points in the training. For example, indicating the end of epochs as vertical lines in the scalar plots to see how many epochs are required to converge.

**Comment by chihuahua** [comment]
Hmm, could you please clarify epochs? The points in scalar charts are already marked by steps:

![ossbgk76y9x-1](https://user-images.githubusercontent.com/4221553/29659810-35e46bd0-8874-11e7-8798-bffc8fb2d05c.png)

but I wasn't sure if that was completely what you meant. Thank you! :)

**Comment by Dapid** [comment]
Sorry for not being clear, I'll try to explain myself. So, every mini batch contributes with one point, and an epoch is a full pass through the data (or an arbitrarily decide number of mini batches). The simplest question that this would help me answer is how many passes through my data do I need to get a good model?

A quick and dirty example (not suggesting this is a good API):

```
for epoch in range(100):
    for minibatch in minibatches:
        sess.run([loss, merged_tb], minibatch)
    merged_tb.mark_time_step(label=' '.format(epoch)) 
```
And that would give me the plot, with a vertical line at the end of each epoch.


But this doesn't have to be just epochs, it can be used to mark points where the learning rate was changed, for example.
